### ISSUE TITLE
Postgres: Add support for ALTER FUNCTION statements

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2089]
<!--If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant.-->

## Expected Behaviour
SQLFluff should be able to parse `ALTER FUNCTION` statements

## Observed Behaviour
```shell
sqlfluff lint --dialect postgres -n sample.sql

== [sample.sql] FAIL                                                                                                                                                                                                                  
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER
                       | FUNCTION fn OWNER TO auser'
```

## Steps to Reproduce
1. Create `sample.sql` using the following query
```sql
ALTER FUNCTION fn OWNER TO auser;
```
2. sqlfluff lint --dialect postgres -n sample.sql
3. sqlfluff parse --dialect postgres -n sample.sql

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2 and master`
`Python 3.8.22`

## Configuration
```
N/A : Default configuration
```
